utils/build-script

```diff
@@ -986,10 +997,48 @@ class BuildScriptInvocation(object):
         #
         # FIXME: This should really be per-host, but the current structure
         # matches that of `build-script-impl`.
-        (impl_product_classes, product_classes) = self.compute_product_classes()
+        (before_impl_product_classes, impl_product_classes, product_classes) =\
+            self.compute_product_classes()
 
         # Execute each "pass".
 
+        # Pre-build-script-impl products...
+        # Note: currently only supports building for the host.
+        for host_target in [self.args.host_target]:
+            for product_class in before_impl_product_classes:
+                if product_class.is_build_script_impl_product():
+                    continue
+                if not product_class.is_before_build_script_impl_product():
+                    continue
+
+                product_source = product_class.product_source_name()
+                product_name = product_class.product_name()
```

Comment:
One thing that we may want to do is refactor out a part of this into a helper.

Reply:
Done.